apps/prairielearn/src/pages/publicQuestions/publicQuestions.ts

```diff
@@ -0,0 +1,35 @@
+import { Router } from 'express';
+import { QuestionsPage } from './publicQuestions.html';
+import { selectPublicQuestionsForCourse } from '../../models/questions';
+import { selectCourseById } from '../../models/course';
+import asyncHandler = require('express-async-handler');
+import { features } from '../../lib/features/index';
+import error = require('@prairielearn/error');
+const router = Router({ mergeParams: true });
+
+router.get(
+  '/',
+  asyncHandler(async function (req, res) {
+    res.locals.course = await selectCourseById(req.params.course_id);
+    const questionSharingEnabled = await features.enabled('question-sharing', {
+      course_id: res.locals.course.id,
+      institution_id: res.locals.course.institution_id,
+    });
+
+    if (!questionSharingEnabled) {
+      throw error.make(404, 'Not Found');
+    }
+
+    const questions = await selectPublicQuestionsForCourse(res.locals.course.id);
+    res.send(
+      QuestionsPage({
+        questions: questions,
+        showAddQuestionButton: false,
+        qidPrefix: '@' + res.locals.course.sharing_name + '/',
```

Comment:
While testing I forgot to set a sharing name and I got the following. If we don't have a sharing name set should we not display the questions table and instead show a message saying "this course doesn't have a sharing name set, do it in the course settings page"?

<img width="331" alt="image" src="https://github.com/PrairieLearn/PrairieLearn/assets/875833/3a0ff454-8d97-4831-a458-69d336149ffe">

Reply:
done, thanks